rfcs/41-ui-navigation.md

```diff
@@ -0,0 +1,364 @@
+# Feature Name: Generic UI Navigation API
+
+## Summary
+
+Introduce [`bevy-ui-navigation`] into the bevy tree.
+
+By default this, amounts to replacing `Interaction` with `Focusable` in `bevy_ui`.
+On top of the current behavior, this adds the following capabilities:
+- Focus management for **gamepads**
+- Decouples input from focus management
+- Decouples focus management from `bevy_ui`
+- User-specified input handling for UI interaction
+- An optional menu tree management system
+
+## Terminology
+
+- _menu tree_: the access hierarchy of menus.
+- _activating_ a focusable: Sending the `Action` event while this focusable is focused.
+- _focused menu_: The menu in which the currently focused `Focusable` is.
+- _child menu_ (of a focusable): The menu accessed from this focusable.
+
+## Motivation
+
+`bevy_ui` interaction story is crude.
+The only way to interact with the UI currently
+is through the `Interaction` and `FocusPolicy` components.
+`Interaction` only supports mouse or touch-based input.
+We want gamepad supports and more game-oriented functionalities,
+such as menu trees and movement between menus.
+
+The current UI focus system is limited to `bevy_ui` and
+can't be swapped out for something the user deems more appropriate.
+
+This is in contradiction with the general philosophy of bevy,
+which is of providing to the user all the tools necessary to create something
+that fits their needs optimally.
+
+This RFC proposes to resolve all those questions.
+[`bevy-ui-navigation`] provides a highly customizable ECS-based navigation engine.
+
+## User-facing explanation
```

Comment:
I think this will probably want a rewrite, but for now the bullet points are fine.

Reply:
I've now copied most of the API docs from bevy-ui-navigation into this RFC. I'm not a fan (just click the "see doc" link :P) but it has the quality of explaining very well each item of the library.